src/api/controllers/ContentController.ts

```diff
@@ -31,63 +33,78 @@ export class ContentController {
-    ContentTransformer.updateOne("contacts", data);
+    @PartialBody() data: GetContentContactsDto
+  ): Promise<GetContentContactsDto> {
+    await ContentTransformer.updateOne("contacts", data);
     return ContentTransformer.fetchOne("contacts");
   }
 
   @Authorized("admin")
   @Patch("/email")
   async updateEmail(
-    @PartialBody() data: GetEmailContentDto
-  ): Promise<GetEmailContentDto> {
-    ContentTransformer.updateOne("email", data);
+    @PartialBody() data: GetContentEmailDto
+  ): Promise<GetContentEmailDto> {
+    await ContentTransformer.updateOne("email", data);
     return ContentTransformer.fetchOne("email");
   }
 
   @Authorized("admin")
   @Patch("/general")
   async updateGeneral(
-    @PartialBody() data: GetGeneralContentDto
-  ): Promise<GetGeneralContentDto> {
-    ContentTransformer.updateOne("general", data);
+    @PartialBody() data: GetContentGeneralDto
+  ): Promise<GetContentGeneralDto> {
+    await ContentTransformer.updateOne("general", data);
     return ContentTransformer.fetchOne("general");
   }
 
   @Authorized("admin")
   @Patch("/join")
   async updateJoin(
-    @PartialBody() data: GetJoinContentDto
-  ): Promise<GetJoinContentDto> {
-    ContentTransformer.updateOne("join", data);
+    @PartialBody() data: GetContentJoinDto
+  ): Promise<GetContentJoinDto> {
+    await ContentTransformer.updateOne("join", data);
     return ContentTransformer.fetchOne("join");
   }
 
   @Authorized("admin")
   @Patch("/join/setup")
   async updateJoinSetup(
-    @PartialBody() data: GetJoinSetupContentDto
-  ): Promise<GetJoinSetupContentDto> {
-    ContentTransformer.updateOne("join/setup", data);
+    @PartialBody() data: GetContentJoinSetupDto
+  ): Promise<GetContentJoinSetupDto> {
+    await ContentTransformer.updateOne("join/setup", data);
     return ContentTransformer.fetchOne("join/setup");
   }
 
   @Authorized("admin")
   @Patch("/profile")
   async updateProfile(
-    @PartialBody() data: GetProfileContentDto
-  ): Promise<GetProfileContentDto> {
-    ContentTransformer.updateOne("profile", data);
+    @PartialBody() data: GetContentProfileDto
+  ): Promise<GetContentProfileDto> {
+    await ContentTransformer.updateOne("profile", data);
     return ContentTransformer.fetchOne("profile");
   }
 
   @Authorized("admin")
   @Patch("/share")
   async updateShare(
-    @PartialBody() data: GetShareContentDto
-  ): Promise<GetShareContentDto> {
-    ContentTransformer.updateOne("share", data);
+    @PartialBody() data: GetContentShareDto
+  ): Promise<GetContentShareDto> {
+    await ContentTransformer.updateOne("share", data);
     return ContentTransformer.fetchOne("share");
   }
+
+  @Authorized("admin")
+  @Patch("/payment")
+  async updatePayment(
+    @PartialBody() data: GetContentPaymentDto
+  ): Promise<GetContentPaymentDto> {
+    if (data.taxRate) {
+      await stripeTaxRateCreateOrRecreateDefault(data.taxRate, {
+        active: data.taxRateEnabled,
+        country: data.stripeCountry
```

Comment:
It would be better to use `config.stripe.country` here as `data.stripeCountry` won't necessarily be set on a partial update, and in any case that's the underlying source of the value

Reply:
What did you think about just setting `data.country ||= config.stripe.country;` in the `stripeTaxRateCreateOrRecreateDefault` method so that config.stripe.country is used if `data.country` undefined? In this way, we retain the option of overwriting the country as well

Reply:
Hmm in my opinion it can be worth trying to solve future problems when it makes for a cleaner architecture, but I don't think this is one of those cases. To really support multiple countries would require a lot more changes, and we don't know what the requirements would be yet, so I would keep it simple for now and not even let `stripeTaxRateCreateOrRecreateDefault` accept a `country`.

Reply:
We can also leave country out completely, the parameter is not absolutely necessary

Reply:
I have now completely rewritten this once again. The tax rate id is now saved in the database and used from there